feat(linter): implement eslint/accessor-pairs rule

tt-a1i opened the pull request:
## Summary

Implements the ESLint `accessor-pairs` rule that enforces getter/setter pairs in objects, classes, and TypeScript interfaces.

### Features
- Checks for setters without getters (default: enabled)
- Checks for getters without setters (default: disabled)
- Supports object literals, class bodies, and Object.defineProperty/defineProperties/create
- TypeScript support for interfaces and type literals (`enforceForTSTypes` option)
- Handles computed property keys using structural comparison via `ContentEq`
- Separates static and instance members in classes

### Configuration options
- `setWithoutGet` (default: true) - Report setters without getters
- `getWithoutSet` (default: false) - Report getters without setters
- `enforceForClassMembers` (default: true) - Apply to class members
- `enforceForTSTypes` (default: false) - Apply to TypeScript interfaces/types

## Test plan
- [x] All 125 pass test cases from ESLint's test suite
- [x] All 177 fail test cases from ESLint's test suite (adapted for TypeScript parser compatibility)
- [x] Snapshot tests generated and verified
- [x] Clippy passes
- [x] Formatting passes

Related to #479 (ESLint core rules umbrella issue)

tt-a1i committed:
feat(linter): implement eslint/accessor-pairs rule

Implements the ESLint `accessor-pairs` rule that enforces getter/setter pairs in objects, classes, and TypeScript interfaces.

## Features
- Checks for setters without getters (default: enabled)
- Checks for getters without setters (default: disabled)
- Supports object literals, class bodies, and Object.defineProperty/defineProperties/create
- TypeScript support for interfaces and type literals (enforceForTSTypes option)
- Handles computed property keys using structural comparison
- Separates static and instance members in classes

## Configuration options
- `setWithoutGet` (default: true) - Report setters without getters
- `getWithoutSet` (default: false) - Report getters without setters
- `enforceForClassMembers` (default: true) - Apply to class members
- `enforceForTSTypes` (default: false) - Apply to TypeScript interfaces/types

Closes #479 (partially)

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

tt-a1i committed:
refactor(linter): improve accessor-pairs following project conventions

- Use DefaultRuleConfig for cleaner configuration parsing
- Use direct imports instead of inline paths (ClassElement)
- Apply rustfmt formatting fixes

connorshea commented on crates/oxc_linter/src/rules/eslint/accessor_pairs.rs:
```diff
@@ -0,0 +1,2074 @@
+            .and_then(|arr| arr.first())
+            .map(|v| serde_json::from_value::<AccessorPairsConfig>(v.clone()).unwrap_or_default())
+            .unwrap_or_default();
+        Self(Box::new(config))
```
Will also need to add the DefaultRuleConfig import, but this should work:

```suggestion
        serde_json::from_value::<DefaultRuleConfig<AccessorPairs>>(value)
            .unwrap_or_default()
            .into_inner()
```

tt-a1i replied:
Done! Used `DefaultRuleConfig<AccessorPairsConfig>` pattern.

connorshea commented on crates/oxc_linter/src/rules/eslint/accessor_pairs.rs:
````diff
@@ -0,0 +1,2074 @@
+    /// ```
+    AccessorPairs,
+    eslint,
+    pedantic,
````
This will fix the test that's failing:

```suggestion
    pedantic,
    config = AccessorPairsConfig,
```

connorshea reviewed (dismissed):
Just a few suggestions

tt-a1i committed:
fix(linter): check global Object/Reflect in accessor-pairs

Add global reference check to avoid false positives when Object or
Reflect is shadowed by a local variable.

Uses ctx.scoping().find_binding() to verify the identifier refers to
the global object.

tt-a1i committed:
refactor(linter): simplify accessor-pairs config

- Use #[serde(default)] at struct level instead of per-field defaults
- Remove default_true helper function
- Add config = AccessorPairsConfig to declare_oxc_lint! macro
- Add doc comments for config options

tt-a1i committed:
chore(linter): regenerate rule runner impls

tt-a1i commented on crates/oxc_linter/src/rules/eslint/accessor_pairs.rs:
```diff
@@ -0,0 +1,2074 @@
+    enforce_for_class_members: bool,
+    #[serde(default, rename = "enforceForTSTypes")]
+    enforce_for_ts_types: bool,
+}
```
Done! Applied the top-level `#[serde(default)]` and added doc comments for each config option.

tt-a1i committed:
fix(linter): remove needless raw string hashes

camc314 committed:
update test cases

camc314 committed:
fix failure

camc314 committed:
remove some allocs

camc314 committed:
u

camc314 reviewed (approved):
Thank you!
